src/core/mac/mac_frame.hpp

```diff
@@ -164,33 +180,182 @@ class ExtAddress: public otExtAddress
+    ShortAddress GetShort(void) const { return mShared.mShortAddress; }
+
+    /**
+     * This method gets the address as an Extended Address.
+     *
+     * This method MUST be used only if the address type is Extended Address.
+     *
+     * @returns A constant reference to the Extended Address.
+     *
+     */
+    const ExtAddress &GetExtended(void) const { return mShared.mExtAddress; }
+
+    /**
+     * This method gets the address as an Extended Address.
+     *
+     * This method MUST be used only if the address type is Extended Address.
+     *
+     * @returns A reference to the Extended Address.
+     *
+     */
+    ExtAddress &GetExtended(void) { return mShared.mExtAddress; }
+
+    /**
+     * This method sets the address with a Short Address.
      *
-     * @param[out]  aBuf   A pointer to the buffer.
+     * The type is also updated to indicate that address is Short.
+     *
+     * @param[in]  aShortAddress  A Short Address
+     *
+     */
+    void SetShort(ShortAddress aShortAddress) { mShared.mShortAddress = aShortAddress; mType = kTypeShort; }
+
+    /**
+     * This method sets the address with an Extended Address.
+     *
+     * The type is also updated to indicate that the address is Extended.
+     *
+     * @param[in]  aExtAddress  An Extended Address
+     *
+     */
+    void SetExtended(const ExtAddress &aExtAddress) { mShared.mExtAddress = aExtAddress; mType = kTypeExtended; }
+
+    /**
+     * This method sets the address with an Extended Address given as byte array.
+     *
+     * The type is also updated to indicate that the address is Extended.
+     *
+     * @param[in]  aBuffer   Pointer to a array containing the Extended Address. `OT_EXT_ADDRESS_SIZE` bytes from buffer
+     *                       are copied to form the Extended Address.
+     * @param[in]  aReverse  If `true` then `OT_EXT_ADDRESS_SIZE` bytes from @p aBuffer are copied in reverse order,
+     *                       otherwise they are copied as provided.
+     *
+     */
+    void SetExtended(const uint8_t *aBuffer, bool aReverse);
+
+    /**
+     * This method indicates whether or not the address is a Short Broadcast Address.
+     *
+     * @returns TRUE if address is Short Broadcast Address, FALSE otherwise.
+     *
+     */
+    bool IsShortAddrBroadcast(void) const { return ((mType == kTypeShort) && (GetShort() == kShortAddrBroadcast)); }
+
+    /**
+     * This method indicates whether or not the address is a Short Invalid Address.
+     *
+     * @returns TRUE if address is Short Invalid Address, FALSE otherwise.
+     *
+     */
+    bool IsShortAddrInvalid(void) const { return ((mType == kTypeShort) && (GetShort() == kShortAddrInvalid)); }
+
+    /**
+     * This method provides assignment operator for Address object.
+     *
+     * @param[in]  aOther  A reference to another Address object.
+     *
+     * @returns A constant reference to the Address.
+     *
+     */
+    const Address &operator=(const Address &aOther);
```

Comment:
Does the assignment operator provide much value?  It seems to marginally increase code size.

Reply:
With a custom `operator=` we get a small enhancement with copying the `uint16_t` if address is short (or type if empty) and not the whole 8 bytes... I agree using default one would be fine. Will change.